roll/__init__.py

```diff
@@ -243,8 +244,12 @@ def __init__(self):
 
     async def __call__(self, request: Request, response: Response):
         try:
+            request.routing, params = self.routes.match(request.path)
             if not await self.hook('request', request, response):
-                params, handler = await self.dispatch(request)
+                if request.method not in request.routing:
+                    raise HttpError(HTTPStatus.METHOD_NOT_ALLOWED)
+                request.kwargs.update(params)
```

Comment:
We may merge `routing` (the route payload) and `kwargs` (the route dynamic variables, if any) at this point. Maybe not in the same dict, in case of key conflicts?
Maybe a `route` object (perf arlert), with two properties `payload` and `vars` ?